infra/docs.sh

```diff
@@ -16,5 +16,5 @@ cp dev/docs/google2ed1af765c529f57.html dev/docs/doc
 
 # Upload the docs.
 if [ "$1" = "--upload" ]; then
-  gsutil -m rsync -d -r dev/docs/doc/ gs://docs.flutter.io/
+  gsutil -m cp -z "js,json,html,css" dev/docs/doc/ gs://docs.flutter.io/
```

Comment:
-z doesn't work with rsync, unfortunately.

We can do a multi-step process to compress locally, then use the gsutil to upload the compressed artifacts with the right metadata, then upload the images without compression and metadata.

Open to ideas here, but I couldn't find a single command to do what we want.